src/nodes/victron-virtual.js

```diff
@@ -824,6 +824,10 @@ module.exports = function (RED) {
                 if (name === 'Settings/Type') propValue = switchType
 
                 iface[key] = propValue !== undefined ? propValue : 0
+
+                if (name === 'Settings/ValidTypes') {
+                  iface[key] = 1 << Number(config[`switch_${i}_type`])
```

Comment:
This is confusing, as `Number(config[`switch_${i}_type`])` is almost the same as `switchType`, except for when it is null or undefined. Is this intentional?

Could this code be improved by making some variable names more specific? `key` is a generic name, maybe something like `switchOutputPropertyKey`?

I don't understand why bit-shifting is the right thing here. Maybe a comment or reference to some place that explains this would help.

Reply:
Fair points. Should be resolved by [this](https://github.com/victronenergy/node-red-contrib-victron/pull/297/commits/252b656e26446683d65768ae4c0ebdc88865bb57) commit.